compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -3036,13 +3036,14 @@ private Long getIntegerLiteral(ParserRuleContext simpleLiteralContext,
     }
 
     private byte[] getBlobLiteral(BallerinaParser.BlobLiteralContext blobLiteralContext) {
+        byte[] blobLiteralValue = null;
         if (blobLiteralContext.Base16BlobLiteral() != null) {
-            return hexStringToByteArray(getBlobTextValue(blobLiteralContext.getText()));
+            blobLiteralValue = hexStringToByteArray(getBlobTextValue(blobLiteralContext.getText()));
         } else if (blobLiteralContext.Base64BlobLiteral() != null) {
-            return Base64.getDecoder().decode(getBlobTextValue(blobLiteralContext.getText()).
+            blobLiteralValue = Base64.getDecoder().decode(getBlobTextValue(blobLiteralContext.getText()).
                     getBytes(StandardCharsets.UTF_8));
         }
-        return null;
+        return blobLiteralValue;
```

Comment:
theoretically, it should be a byte[0]; but not null, even though this logic never executes with current grammar.

Reply:
Changed to empty array as initialized value. But as mentioned, that logic will never be executed as blob literal will either be base16 or base64.